cmds/core/ls/ls_test.go

```diff
@@ -5,147 +5,230 @@
 package main
 
 import (
+	"bytes"
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
 
 	"github.com/u-root/u-root/pkg/ls"
-	"github.com/u-root/u-root/pkg/testutil"
 )
 
-var tests = []struct {
-	flags []string
-	out   string
-}{
-	{
-		flags: []string{},
-		out: `d1
-f1
-f2
-f3?line 2
-`,
-	}, {
-		flags: []string{"-Q"},
-		out: `"d1"
-"f1"
-"f2"
-"f3\nline 2"
-`,
-	}, {
-		flags: []string{"-aR"},
-		out: `.
-.f4
-d1
-d1/f4
-f1
-f2
-f3?line 2
-`,
-	}, {
-		flags: []string{"-R"},
-		out: `d1
-d1/f4
-f1
-f2
-f3?line 2
-`,
-	}, {
-		flags: []string{"-a"},
-		out: `.
-.f4
-d1
-f1
-f2
-f3?line 2
-`,
-	}, {
-		flags: []string{"f1", "d1", "f2"},
-		out: `f1
-d1:
-f4
-f2
-`,
-	}, {
-		flags: []string{"-d"},
-		out: `.
-`,
-	}, {
-		flags: []string{"-d", "f1", "d1", "f2"},
-		out: `f1
-d1
-f2
-`,
-	}, {
-		flags: []string{"-S"},
-		out: `f2
-d1
-f1
-f3?line 2
-`,
-	},
+type ttflags struct {
+	all       bool
+	human     bool
+	directory bool
+	long      bool
+	quoted    bool
+	recurse   bool
+	classify  bool
 }
 
-func TestLs(t *testing.T) {
+// Test listName func
+func TestListName(t *testing.T) {
+	// Create some directorys.
 	tmpDir := t.TempDir()
-
-	// Create an empty directory.
 	testDir := filepath.Join(tmpDir, "testDir")
-	if err := os.Mkdir(testDir, 0o700); err != nil {
-		t.Fatal(err)
+	os.Mkdir(testDir, 0o700)
```

Comment:
Don't know how I missed this, but you need to use ioutil.TempDir, to ensure two people running this on the same machine don't clash.

Reply:
Use `t.TempDir()`